synapse/handlers/receipts.py

```diff
@@ -188,15 +188,22 @@ def filter_out_hidden(events: List[JsonDict], user_id: str) -> List[JsonDict]:
 
                 new_users = {}
                 for rr_user_id, user_rr in m_read.items():
-                    hidden = user_rr.get("hidden", None)
-                    if hidden is not True or rr_user_id == user_id:
+                    try:
+                        hidden = user_rr.get("hidden", None)
+                        if hidden is not True or rr_user_id == user_id:
+                            new_users[rr_user_id] = user_rr.copy()
+                            # If hidden has a value replace hidden with the correct prefixed key
+                            if hidden is not None:
+                                new_users[rr_user_id].pop("hidden")
+                                new_users[rr_user_id][
+                                    ReadReceiptEventFields.MSC2285_HIDDEN
+                                ] = hidden
+                    except AttributeError:
+                        # Due to https://github.com/matrix-org/synapse/issues/10376
+                        # there are cases where user_rr is a string, in those cases
+                        # we just copy the user_rr
                         new_users[rr_user_id] = user_rr.copy()
```

Comment:
Perhaps we should just ignore these read receipts or delete their data? I am not completely sure what is the best thing to do here